src/domains/user/pages/instructor/InstructorCourses.jsx

```diff
@@ -1,7 +1,34 @@
-import React from "react";
+import { useAuthState } from "@/domains/auth/hooks/useAuthState";
 import styles from "@/domains/user/pages/MyPageSections.module.css";
+import { getInstructorCourses } from "@/domains/user/services/instructorService";
+import { useEffect, useState } from "react";
+import { NavLink } from "react-router";
 
 export default function InstructorCourses() {
+  const { user, loading: userLoading } = useAuthState();
+  const [courses, setCourses] = useState([]);
+  const [coursesLoading, setCoursesLoading] = useState(false);
+
+  // user.id 준비된 뒤 → 데이터 가져오기
+  useEffect(() => {
+    if (userLoading || !user?.id) return;
+    setCoursesLoading(true);
+
+    getInstructorCourses(user.id)
+      .then((data) => setCourses(data))
+      .finally(() => setCoursesLoading(false));
+  }, [user, userLoading]);
```

Comment:
```suggestion
  // user.id 준비된 뒤 → 데이터 가져오기
  useEffect(() => {
    if (userLoading || !user?.id) return;
    setCoursesLoading(true);

    getInstructorCourses(user.id)
      .then(setCourses)
      .catch((err) => {
        console.error(err);
      })
      .finally(() => setCoursesLoading(false));
  }, [user?.id, userLoading]);
```

에러 핸들링 추가하고 의존성 배열 리렌더링 적게 바꿔봤습니다!